components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/AppRoleMappingConfig.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.apache.axiom.om.OMElement;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Application role mapping type of application for an IdP.
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "ApplicationRoleMappingConfig")
+public class AppRoleMappingConfig implements Serializable {
+
+    private static final long serialVersionUID = -1870576785207238676L;
+    private static final String IDP_NAME = "IdPName";
+    private static final String USE_APP_ROLE_MAPPINGS = "UseAppRoleMappings";
+
+    @XmlElement(name = IDP_NAME)
+    private String idPName;
+
+    @XmlElement(name = USE_APP_ROLE_MAPPINGS)
+    private boolean useAppRoleMappings;
+
+    /**
+     * Build ApplicationRoleMappingConfig from OMElement which indicates whether the IdP will use mappings from
+     * idp-roles to application roles for the application.
+     *
+     * @param applicationRoleMappingConfigOM OMElement
+     * @return ApplicationRoleMappingConfig
+     */
+    public static AppRoleMappingConfig build(OMElement applicationRoleMappingConfigOM) {
+
+        AppRoleMappingConfig applicationRoleMappingType = new AppRoleMappingConfig();
+        Iterator<?> iterator = applicationRoleMappingConfigOM.getChildElements();
+
+        while (iterator.hasNext()) {
+            OMElement omElement = (OMElement) iterator.next();
+            if (IDP_NAME.equals(omElement.getLocalName())) {
+                applicationRoleMappingType.setIdPName(omElement.getText());
+            } else if (USE_APP_ROLE_MAPPINGS.equals(omElement.getLocalName())) {
+                applicationRoleMappingType.setUseAppRoleMappings(Boolean.parseBoolean(omElement.getText()));
+            }
+        }
+        return applicationRoleMappingType;
+    }
+
+    /**
+     * Returns the name of the federated IdP used for application role mapping.
```

Comment:
```suggestion
     * Return the name of the federated IdP used for application role mapping.
```

Reply:
Fix other places as well. Make the verb prural

Reply:
Addressed in 4cdf9672bdd6a4d6060995024d14bd9dc79a1168